glusterd2/commands/volumes/volume-option.go

```diff
@@ -91,9 +134,26 @@ func volumeOptionsHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	lock, unlock, err := transaction.CreateLockSteps(volname)
+	lock, unlock := transaction.CreateLockFuncs(volname)
```

Comment:
Since this is such a repeated pattern, I'm thinking of extending the transaction framework to make this simpler.
Like so,
```go
// NewTxnWithLock creates a Txn with the provided context, and immediately obtains cluster locks on the provided lockIDS, and attaches the obtained locks to the Txn
func NewTxnWithLock(ctx context.Context, lockIDS ...string) (*Txn, error)
// Lock obtains cluster locks on the provided lockID (and lockIDS), and attaches the obtained locks to the Txn
func (t *Txn)Lock(lockIDS ...string) error
// Done frees up any held locks and cleans up the Txn
// Note: Done replaces Cleanup
func (t *Txn)Done
```

This would create the following pattern for transactions that require locks.
```go

initialLockIDS = parseReq(req)
// Haven't yet looked up the actual resources in the store
txn, _ = transaction.NewTxnWithLocks(ctx, initialLockIDS...)
defer txn.Done()

// Lookup resources in store
moreLockIDs = parseResourcesAndGetLockIDS()
_ = txn.Lock(moreLockIDs)

// Setup rest of the transaction and perform the actual transaction
txn.Steps = []transaction.Step{...}
_ = txn.Do()
```
Transactions that don't require locks continue as before.

How does this sound?

Reply:
Sounds good to me. However, we need the ability to take such locks in handlers that do not necessarily need a transaction, for example updating peer metadata.